File: src/main/java/com/esprit/models/Post.java
package com.esprit.models;

import java.time.LocalDate;  // Importation de LocalDate

public class Post {
    private int id;
    private String author;
    private String contenu;
    private int upvotes;
    private int downvotes;
    private String image;
    private LocalDate date;  // Modification du type en LocalDate
    private int client_id;

    // Constructeur pour recherche
    public Post(int id, String author, String contenu, int upvotes, int downvotes, String image, LocalDate date, int client_id) {
        this.id = id;
        this.author = author;
        this.contenu = contenu;
        this.upvotes = upvotes;
        this.downvotes = downvotes;
        this.image = image;
        this.date = date;
        this.client_id = client_id;
    }

    // Constructeur pour ajouter
    public Post(String author, String contenu, int upvotes, int downvotes, String image, LocalDate date, int client_id) {
        this.author = author;
        this.contenu = contenu;
        this.upvotes = upvotes;
        this.downvotes = downvotes;
        this.image = image;
        this.date = date;
        this.client_id = client_id;
    }

    // Constructeur pour modification et suppression
    public Post(int id, String author, String contenu, int upvotes, int downvotes, String image) {
        this.id = id;
        this.author = author;
        this.contenu = contenu;
        this.upvotes = upvotes;
        this.downvotes = downvotes;
        this.image = image;
    }

    // Getters et Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getAuthor() { return author; }
    public void setAuthor(String author) { this.author = author; }

    public String getContenu() { return contenu; }
    public void setContenu(String contenu) { this.contenu = contenu; }

    public int getUpvotes() { return upvotes; }
    public void setUpvotes(int upvotes) { this.upvotes = upvotes; }

    public int getDownvotes() { return downvotes; }
    public void setDownvotes(int downvotes) { this.downvotes = downvotes; }

    public String getImage() { return image; }
    public void setImage(String image) { this.image = image; }

    public LocalDate getDate() { return date; }  // Modification du getter
    public void setDate(LocalDate date) { this.date = date; }  // Modification du setter

    public int getClientId() { return client_id; }
    public void setClientId(int client_id) { this.client_id = client_id; }

    @Override
    public String toString() {
        return "Post{" +
                "id=" + id +
                ", author='" + author + '\'' +
                ", contenu='" + contenu + '\'' +
                ", upvotes=" + upvotes +
                ", downvotes=" + downvotes +
                ", image='" + image + '\'' +
                ", date=" + date +  // Affichage de la date comme LocalDate
                ", client_id=" + client_id +
                '}';
    }
}

File: src/Entity/Post.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

use App\Entity\Client;
use Doctrine\Common\Collections\Collection;
use App\Entity\Commentaire;

#[ORM\Entity]
class Post
{

    #[ORM\Id]
    #[ORM\Column(type: "integer")]
    private int $id;

        #[ORM\ManyToOne(targetEntity: Client::class, inversedBy: "posts")]
    #[ORM\JoinColumn(name: 'client_id', referencedColumnName: 'id', onDelete: 'CASCADE')]
    private Client $client_id;

    #[ORM\Column(type: "date")]
    private \DateTimeInterface $date;

    #[ORM\Column(type: "string", length: 250)]
    private string $author;

    #[ORM\Column(type: "string", length: 1000)]
    private string $contenu;

    #[ORM\Column(type: "integer")]
    private int $upvotes;

    #[ORM\Column(type: "integer")]
    private int $downvotes;

    #[ORM\Column(type: "string", length: 1000)]
    private string $image;

    public function getId()
    {
        return $this->id;
    }

    public function setId($value)
    {
        $this->id = $value;
    }

    public function getClient_id()
    {
        return $this->client_id;
    }

    public function setClient_id($value)
    {
        $this->client_id = $value;
    }

    public function getDate()
    {
        return $this->date;
    }

    public function setDate($value)
    {
        $this->date = $value;
    }

    public function getAuthor()
    {
        return $this->author;
    }

    public function setAuthor($value)
    {
        $this->author = $value;
    }

    public function getContenu()
    {
        return $this->contenu;
    }

    public function setContenu($value)
    {
        $this->contenu = $value;
    }

    public function getUpvotes()
    {
        return $this->upvotes;
    }

    public function setUpvotes($value)
    {
        $this->upvotes = $value;
    }

    public function getDownvotes()
    {
        return $this->downvotes;
    }

    public function setDownvotes($value)
    {
        $this->downvotes = $value;
    }

    public function getImage()
    {
        return $this->image;
    }

    public function setImage($value)
    {
        $this->image = $value;
    }

    #[ORM\OneToMany(mappedBy: "post_id", targetEntity: Vote::class)]
    private Collection $votes;

        public function getVotes(): Collection
        {
            return $this->votes;
        }
    
        public function addVote(Vote $vote): self
        {
            if (!$this->votes->contains($vote)) {
                $this->votes[] = $vote;
                $vote->setPost_id($this);
            }
    
            return $this;
        }
    
        public function removeVote(Vote $vote): self
        {
            if ($this->votes->removeElement($vote)) {
                // set the owning side to null (unless already changed)
                if ($vote->getPost_id() === $this) {
                    $vote->setPost_id(null);
                }
            }
    
            return $this;
        }

    #[ORM\OneToMany(mappedBy: "post_id", targetEntity: Commentaire::class)]
    private Collection $commentaires;

    public function __construct()
    {
        $this->votes = new ArrayCollection();
        $this->commentaires = new ArrayCollection();
    }

    public function getClientId(): ?Client
    {
        return $this->client_id;
    }

    public function setClientId(?Client $client_id): static
    {
        $this->client_id = $client_id;

        return $this;
    }

    /**
     * @return Collection<int, Commentaire>
     */
    public function getCommentaires(): Collection
    {
        return $this->commentaires;
    }

    public function addCommentaire(Commentaire $commentaire): static
    {
        if (!$this->commentaires->contains($commentaire)) {
            $this->commentaires->add($commentaire);
            $commentaire->setPostId($this);
        }

        return $this;
    }

    public function removeCommentaire(Commentaire $commentaire): static
    {
        if ($this->commentaires->removeElement($commentaire)) {
            // set the owning side to null (unless already changed)
            if ($commentaire->getPostId() === $this) {
                $commentaire->setPostId(null);
            }
        }

        return $this;
    }
}
